core/src/test/java/kafka/admin/DescribeAuthorizedOperationsTest.java

```diff
@@ -0,0 +1,275 @@
+        return createAdminConfig(username, password, Map.of());
+    }
+
+    private Map<String, Object> createAdminConfig(String username, String password, Map<String, Object> additionalConfigs) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_PLAINTEXT.name);
+        configs.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
+        configs.put(SaslConfigs.SASL_JAAS_CONFIG,
+            String.format("org.apache.kafka.common.security.plain.PlainLoginModule required username=\"%s\" password=\"%s\";", username, password));
+        configs.putAll(additionalConfigs);
+        return configs;
+    }
+
+    private void setupSecurity(ClusterInstance clusterInstance) {
+        try (Admin admin = clusterInstance.admin(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD))) {
+            ResourcePattern topicResource = new ResourcePattern(ResourceType.TOPIC, AclEntry.WILDCARD_RESOURCE, PatternType.LITERAL);
+
+            CreateAclsResult result = admin.createAcls(List.of(
+                new AclBinding(CLUSTER_PATTERN, ALTER_ENTRY),
+                new AclBinding(topicResource, DESCRIBE_ENTRY)
+            ));
+            assertDoesNotThrow(result::all);
+
+            assertDoesNotThrow(() -> clusterInstance.waitAcls(new AclBindingFilter(CLUSTER_PATTERN.toFilter(), ANY), Set.of(ALTER_ENTRY)));
+            assertDoesNotThrow(() -> clusterInstance.waitAcls(new AclBindingFilter(topicResource.toFilter(), ANY), Set.of(DESCRIBE_ENTRY)));
+        }
+    }
+
+    @ClusterTemplate("generator")
+    public void testConsumerGroupAuthorizedOperations(ClusterInstance clusterInstance) {
+        setupSecurity(clusterInstance);
+        try (Admin admin = clusterInstance.admin(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD))) {
+            assertDoesNotThrow(() -> admin.createTopics(List.of(new NewTopic("topic1", 1, (short) 1))));
+            assertDoesNotThrow(() -> clusterInstance.waitForTopic("topic1", 1));
+        }
+        try (Admin admin = clusterInstance.admin(createAdminConfig(JaasUtils.KAFKA_PLAIN_USER1, JaasUtils.KAFKA_PLAIN_USER1_PASSWORD));
+             Consumer<Byte, Byte> consumer1 = clusterInstance.consumer(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD, Map.of(ConsumerConfig.GROUP_ID_CONFIG, GROUP1)));
+             Consumer<Byte, Byte> consumer2 = clusterInstance.consumer(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD, Map.of(ConsumerConfig.GROUP_ID_CONFIG, GROUP2)));
+             Consumer<Byte, Byte> consumer3 = clusterInstance.consumer(createAdminConfig(JaasUtils.KAFKA_PLAIN_ADMIN, JaasUtils.KAFKA_PLAIN_ADMIN_PASSWORD, Map.of(ConsumerConfig.GROUP_ID_CONFIG, GROUP3)))
+        ) {
+            // create consumers to avoid group not found error
+            consumer1.subscribe(List.of("topic1"));
```

Comment:
Could we use `Admin.alterConsumerGroupOffsets()` to create the consumer groups?

Reply:
Updated it. Thanks.